CONTRIBUTING.md

```diff
@@ -0,0 +1,181 @@
+
+### Patch requirements
+
+- Maintainers MUST have a Platform account and SHOULD use their real names or a well-known alias.
+- Contributors SHOULD have a Platform account and MAY use their real names or a well-known alias.
+- A patch SHOULD be a minimal and accurate answer to exactly one identified and agreed problem.
+- A patch MUST adhere to the code style guidelines of the project if these are defined.
+- A patch MUST adhere to the "Evolution of Public Contracts" guidelines defined below.
+- A patch MUST NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+- A patch MUST compile cleanly and pass project self-tests on at least the principle target platform.
+- A patch commit message SHOULD consist of a single short (less than 50 character) line summarizing the change, optionally followed by a blank line and then a more thorough description.
+- A "Correct Patch" is one that satisfies the above requirements.
+
+### Development process
+
+- Change on the project MUST be governed by the pattern of accurately identifying problems and applying minimal, accurate solutions to these problems.
+- To request changes, a user SHOULD log an issue on the project Platform issue tracker.
+- The user or Contributor SHOULD write the issue by describing the problem they face or observe.
+- The user or Contributor SHOULD seek consensus on the accuracy of their observation, and the value of solving the problem.
+- Users MUST NOT log feature requests, ideas, or suggestions unrelated to Monero code or Monero's dependency code or Monero's potential/future dependency code or research which successfully implements Monero.
+- Users MUST NOT log any solutions to problems (verifiable or hypothetical) of which are not explicitly documented and/or not provable and/or cannot be reasonably proven.
+- Thus, the release history of the project MUST be a list of meaningful issues logged and solved.
+- To work on an issue, a Contributor MUST fork the project repository and then work on their forked repository.
+- To submit a patch, a Contributor MUST create a Platform pull request back to the project.
+- A Contributor MUST NOT commit changes directly to the project.
+- To discuss a patch, people MAY comment on the Platform pull request, on the commit, or elsewhere.
+- To accept or reject a patch, a Maintainer MUST use the Platform interface.
+- Maintainers SHOULD NOT merge their own patches except in exceptional cases, such as non-responsiveness from other Maintainers for an extended period (more than 30 days) or unless urgent as defined by the Monero Maintainers Team.
+- Maintainers MUST NOT make value judgments on correct patches unless the Maintainer (as may happen in rare circumstances) is a core code developer.
+- Maintainers MUST NOT merge pull requests in less than 168 hours (1 week) unless deemed urgent by at least 2 people from the Monero Maintainer Team.
```

Comment:
Is there a good reason to delay a merge by up to one week if the PR has gained sufficient approvals? Would it not make more sense to simply gate merges by some number of approvals from contributors?

Less-than-trivial changes naturally take more time to review.

Reply:
I think this should be kept in so that people can get a chance to add their opinion to proposed changes. For pure CI changes it can be skipped IMO.

Reply:
The sentence, as written, makes no sense, as it would mean the act of merging itself cannot take less than 168 hours. Moreover, would the 168 hour delay apply from the time of submitting the patch, or gaining the requisite approvals? In any case, I don't like the idea, as it incentivizes larger PRs purely for the sake of avoiding lengthy merge delays.

Reply:
Regardless of the exact interpretation of this policy, I dislike it: Assuming an even marginally effective review by those who sign off on the merge, the merged code should net improve the software. Delaying an improvement on the prospect that someone might later object to it doesn't make sense: A patch, or comment, subsequent to the merge can correct any perceived deficiency. And if one objects to the assumption of competent reviewers, then the merge delay may as well be zero, because the review would not improve the work.